core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -742,6 +745,40 @@ object Parser {
         }
       }
 
+    implicit val catsShowExpectation: Show[Expectation] =
+      new Show[Expectation] {
+        private val dq = "\""
+        def show(expectation: Expectation): String = expectation match {
+          case OneOfStr(_, strs: List[String]) =>
+            "one of: " + strs.map(s => dq + s + dq).mkString("{", ", ", "}")
+
+          case InRange(_, lower: Char, upper: Char) =>
+            if (lower != upper) s"in range: ['$lower', '$upper']"
+            else s"is: '$lower'"
+
+          case StartOfString(_) =>
+            "start of string"
+
+          case EndOfString(_, length) =>
+            s"end of string, length = $length"
```

Comment:
I think `end of string, but found $length chars remaining` or something might be better. Length is unclear as an error message. Or, you can simply write `end of string` if you are already pointing to the location is may be pretty obvious how much is left.